### PR TITLE
chore(example): update to vitejs

### PR DESCRIPTION
**Pull request by Francois-Esquire**
## Description

Updates the `example/` directory to use `vitejs` and support `node.js` v18.

## Detail

- Bundling has migrated from `create-react-app` to [`vite`](https://vitejs.dev/)
- `yarn create vite example` was used as the basis to migrate over the example

### CODE REVIEW

File: example/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  "parserOptions": {
+    "requireConfigFile": false
+  },
+  "overrides": [
+    {
+      "files": ["*.ts", "*.tsx"],
+      "extends": ["@zendeskgarden/eslint-config/plugins/typescript", "prettier"]
+    }
+  ]
+}
```
Comment (Francois-Esquire) [inline example/.eslintrc.json:12]: It may be worthwhile to move eslint to the root of the project instead of have it nested individually.

**Comment by geotrev** [comment]
Are there any trade offs vs. other app bundlers? Or is the goal just to have something relatively straightforward to replace CRA due to its incompatibility with node LTS?

**Comment by Francois-Esquire** [comment]
> Are there any trade offs vs. other app bundlers? Or is the goal just to have something relatively straightforward to replace CRA due to its incompatibility with node LTS?

@geotrev Ultimately, it is a drop in replacement for CRA for compatibility however it does bring simplicity compared to other bundlers with some configurability if needed. Vite also uses `rollup` under the hood which is a common tool in our toolchain.

**Review by geotrev — APPROVED** [review]
Demo runs well and no major comments except that you already know I think Vite is great. I like the simplicity it creates!

File: example/.eslintrc.json (hunk 1 above)
Comment (jzempel) [reply example/.eslintrc.json:12]: Can we capture this as a follow-on?

File: example/package.json [hunk 2]
```diff
@@ -4,42 +4,50 @@
...
     "lint": "yarn lint:js && yarn lint:md",
-    "lint:js": "eslint src/ --ext js,ts,tsx --max-warnings 0",
+    "lint:js": "eslint . --ext js,ts,tsx",
     "lint:md": "markdownlint README.md",
     "prepare": "yarn build",
-    "start": "react-scripts start",
-    "test:ci": "yarn format --check && yarn lint"
+    "start": "yarn dev",
+    "test:ci": "yarn format --check && yarn lint",
+    "preview": "vite preview"
```
Comment (jzempel) [inline example/package.json:17]: Can we stick with standard Garden scripting and parity with previous?
- modify to `"start": "vite --open"`
- no `dev`
- no `preview` (similar to `website` where rare production run can be executed as needed)

**Review by jzempel — CHANGES_REQUESTED** [review]

File: example/package.json (hunk 2 above)
Comment (Francois-Esquire) [reply example/package.json]: Sure, I've updated in [fb65a8a](https://github.com/zendeskgarden/tailwindcss/pull/150/commits/fb65a8a3b0c4ac835cdea1fd6902b5269c2ddf29).

**Review by jzempel — APPROVED** [review]

**Pull request merged**